book/src/proposals/leader-duplicate-block-slashing.md

```diff
@@ -0,0 +1,70 @@
+# Leader Duplicate Block Slashing
+
+This design describes how the cluster slashes leaders that produce
+duplicate blocks.
+
+Leaders that produce multiple blocks for the same slot increase the
+number of potential forks that the cluster has to resolve.
+
+## Overview
+
+1. Slashing Condition: If you have voted on version `S1` of a slot, it is a slashing condition to vote on any descendant of another version `S2` of the slot, unless that descendant of version `S2` has seen supermajority (> 66%) of the stake voting on it (See section `Slashing Condition` below for how this proof is to be generated).
+2. If you see two blockhashes for the same block, and you are less than 2^THRESHOLD lockout on that block, drop it. Flag the block as "duplicated".
+3. Only repair a version of the "duplicated" block if you see a `Repair Proof` as outlined in the `Repair Proof` section below
+
+## Primitives
+
+For a bank `B`, let `A` be the latest ancestor of `B` that has gotten supermajority > 66% votes. 
```

Comment:
@carllin latest ancestor of `B` that has gotten supermajority, can likely be different for different validators when we are calculating `bank_hash` (depending upon network topology/delay), and also likely to change once ancestor of `B` which is `> A` achieves supermajority, later on. 

Also, Is it a good idea to add consensus variables to ledger?

Reply:
@ParthDesai its the latest ancestor off `B` that has gotten supermajority based on the vote states in `B`, so it should be consistent.